xbmc/cores/FFmpeg.cpp

```diff
@@ -21,6 +21,19 @@
 
 static thread_local CFFmpegLog* CFFmpegLogTls;
 
+namespace FFMPEG_HELP_TOOLS
+{
+
+std::string FFMpegErrorToString(int err)
+{
+  std::string text;
+  text.reserve(AV_ERROR_MAX_STRING_SIZE);
+  av_strerror(err, text.data(), AV_ERROR_MAX_STRING_SIZE);
+  return text;
```

Comment:
I know this code already existed before but anyway it doesn't work because the `std::string` length is still zero: https://godbolt.org/z/P3eKTfWGz
I think the only way to fix this is to allocate a `char[AV_ERROR_MAX_STRING_SIZE]` on the stack and assign the `std::string` from there.

Reply:
I think the solution is to use `resize` rather than `reserve`.

https://en.cppreference.com/w/cpp/string/basic_string/resize

Reply:
`AV_ERROR_MAX_STRING_SIZE` is only 64 so I think it's ok:

```c++
  text.resize(AV_ERROR_MAX_STRING_SIZE);
  av_strerror(err, text.data(), text.size());
```

If it were a much larger number (1000, 2000) then it would be better to do what neo1973 says to avoid always returning a string of maximum length.